HopsanGUI/Widgets/TextEditorWidget.cpp

```diff
@@ -399,17 +399,47 @@ void TextEditor::keyPressEvent(QKeyEvent* event)
     //Replace tabs with four whitespaces
     if(event->key() == Qt::Key_Backtab)    //Shift+tab, remove spaces to the left until next tab stop (or as many spaces as possible)
     {
-        QString line = this->textCursor().block().text().toLatin1();
-        int pos = textCursor().positionInBlock();
-        while(pos > 0 && line.at(pos-1) == ' ')
+        if(textCursor().anchor() != textCursor().position())    //Selection exists, indent whole selection
         {
-            this->textCursor().deletePreviousChar();
-            line = this->textCursor().block().text().toLatin1();
-            pos = textCursor().positionInBlock();
-            if(pos%2 == 0)
-            {
-                break;
+            auto cursor = textCursor();
+            QString text = cursor.selection().toPlainText();
+            if(text.startsWith(" ")) {
+                text.remove(0,1);
+            }
+            if(text.startsWith(" ")) {
+                text.remove(0,1);
             }
+            text.replace("\n ","\n");
+            text.replace("\n ","\n");
```

Comment:
Why do this twice? it is unclear to me what this is for, something with indentation I assume

Reply:
We want to remove two spaces at the beginning, but if there is only one spaces we only want to remove one. Therefore the easiest solution is to iteratively remove one space at a time. If we want to change tab width later we could replace this by a loop.